apps/frontend/src/app/3_organisms/TransactionStepDialog/components/TransactionStep/TransactionStep.tsx

```diff
@@ -0,0 +1,255 @@
+      helper:
+        'Limiting the amount of approved tokens as an additional security measure may result higher fees',
+    },
+  ];
+
+  const [advanced, setAdvanced] = useState(false);
+  const onChange = useCallback(
+    e => {
+      updateConfig({
+        ...config,
+        unlimitedAmount: e.target.value === 'unlimited_amount',
+      });
+    },
+    [config, updateConfig],
+  );
+  const disabledSettings =
+    isLoading || ![StatusType.idle, StatusType.error].includes(status);
+
+  const estimatedGasFee = useMemo(() => {
+    return config.gasLimit && config.gasPrice
+      ? bignumber(config.gasPrice?.toString())
+          .mul(config.gasLimit?.toString())
+          .div(10 ** 9)
+          .toFixed(8)
+      : '';
+  }, [config.gasLimit, config.gasPrice]);
+
+  return (
+    <div className="flex flex-col">
+      <StatusItem content={step} label={title} status={status} />
+      <div className="ml-10">
+        {status === StatusType.error && (
+          <Paragraph className="text-error-light">
+            Your transaction has failed. <br />
+            Please close or retry your transaction
+          </Paragraph>
+        )}
+        {subtitle && status !== StatusType.error && (
+          <Paragraph className="text-gray-30">{subtitle}</Paragraph>
+        )}
+        <SimpleTable className="max-w-72 mt-3">
+          {config.amount !== undefined && (
+            <SimpleTableRow
+              label="Amount"
+              value={`${config.unlimitedAmount ? '∞' : parsedAmount} ${
+                token?.symbol
+              }`}
+              valueClassName="text-primary-10"
+            />
+          )}
+          <SimpleTableRow
+            label={
+              <span className="flex items-center">
+                Estimated gas fee
+                <HelperButton className="ml-1.5" content="Estimated gas fee" />
+              </span>
+            }
+            value={estimatedGasFee + ' rBTC'}
+            valueClassName="text-primary-10"
+          />
+          {config.hash && (
+            <SimpleTableRow
+              label="TX ID"
+              value={
+                <TransactionId
+                  href={`${chain?.blockExplorerUrl}/${config.hash}`}
+                  value={config.hash}
+                />
+              }
+            />
+          )}
+        </SimpleTable>
+
+        <Accordion
+          className="mt-4 text-xs"
+          label="Advanced Settings"
+          open={advanced && !disabledSettings}
+          onClick={() => setAdvanced(!advanced)}
+          disabled={disabledSettings}
+        >
+          {config.amount !== undefined && (
+            <>
+              <RadioButtonGroup
+                options={amountOptions}
+                onChange={onChange}
+                className="mt-1"
+                defaultChecked={config.unlimitedAmount ? 1 : 0}
+              />
+              <Heading type={HeadingType.h3} className="mb-3">
+                Approval gas setting
```

Comment:
Copy should be changed to something generic I think. "Approval" assumes user is in the token approval step, maybe should be just "Gas settings".